app/src/main/kotlin/com/hedvig/app/feature/settings/SettingsViewModel.kt

```diff
@@ -2,29 +2,42 @@ package com.hedvig.app.feature.settings
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.hedvig.android.apollo.graphql.type.Locale
-import com.hedvig.app.feature.marketpicker.LanguageRepository
-import com.hedvig.app.feature.marketpicker.LocaleBroadcastManager
+import com.apollographql.apollo3.ApolloClient
+import com.hedvig.android.apollo.graphql.UpdateLanguageMutation
+import com.hedvig.android.market.Language
+import com.hedvig.app.LanguageService
 import com.hedvig.app.util.apollo.NetworkCacheManager
+import com.hedvig.app.util.apollo.safeQuery
 import com.hedvig.hanalytics.AppScreen
 import com.hedvig.hanalytics.HAnalytics
 import kotlinx.coroutines.launch
 
 class SettingsViewModel(
-  private val repository: LanguageRepository,
-  private val localeBroadcastManager: LocaleBroadcastManager,
   hAnalytics: HAnalytics,
-  private val cacheManager: NetworkCacheManager,
+  private val changeDeviceLanguageAndUploadPreferredLanguageUseCase: ChangeDeviceLanguageAndUploadPreferredLanguageUseCase,
 ) : ViewModel() {
   init {
     hAnalytics.screenView(AppScreen.APP_SETTINGS)
   }
 
-  fun save(acceptLanguage: String, locale: Locale) {
+  fun applyLanguage(language: Language) {
     viewModelScope.launch {
-      repository.uploadLanguage(acceptLanguage, locale)
+      changeDeviceLanguageAndUploadPreferredLanguageUseCase.invoke(language)
     }
+  }
+}
+
+class ChangeDeviceLanguageAndUploadPreferredLanguageUseCase(
+  private val apolloClient: ApolloClient,
+  private val languageService: LanguageService,
+  private val cacheManager: NetworkCacheManager,
+) {
+  suspend fun invoke(language: Language) {
+    apolloClient
+      .mutation(
+        UpdateLanguageMutation(language.toString(), language.toLocale()),
```

Comment:
Just a generic question, but what is the reason this mutation exists in the first place? Is it that the backend is somehow storing our language preference as well on top of doing it in app? Is this a result of us not using a `Accept-language` header everywhere in the backend therefore it needs to have this stored somewhere? I am thinking there's just some historical knowledge here that you may have.

Reply:
It is for the backend to know in certain contexts which language to present information, but I'm not quite sure which still use it, if any.

`Accept-Language` as a standard would indeed probably be better.